azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/AvailabilitySetImpl.java

```diff
@@ -136,4 +144,22 @@ protected void createResource() throws Exception {
         this.idOfVMsInSet = null;
         this.vmsInSet = null;
     }
+
+    @Override
+    protected ServiceCall createResourceAsync(final ServiceCallback<Void> callback) {
+        return this.client.createOrUpdateAsync(this.resourceGroupName(), this.name(), this.inner(),
+                Utils.fromVoidCallback(this, new ServiceCallback<Void>() {
+                    @Override
+                    public void failure(Throwable t) {
```

Comment:
Neat. One minor thing maybe is the callback naming. Nouns as method names (that are not merely simple property accessors) seem a little bit odd. Not sure if there is official guidance on this, but onFailure, onSuccess sounds a bit more intuitive to me.  (example usage: http://www.gwtproject.org/javadoc/latest/com/google/gwt/user/client/rpc/AsyncCallback.html)